crates/examples/src/bin/segments.rs

```diff
@@ -0,0 +1,61 @@
+use object::{
+    Object, ObjectSegment, SEGMENT_RWE_FLAG_EXECUTE, SEGMENT_RWE_FLAG_READ, SEGMENT_RWE_FLAG_WRITE,
+};
+use std::{env, fs, process};
+
+fn main() {
+    let mut args = env::args().skip(1);
+    if args.len() < 1 {
+        eprintln!("Usage: {} <object file path>", env::args().next().unwrap());
+        process::exit(1);
+    }
+
+    let file_path = args.next().unwrap();
+
+    let file = match fs::File::open(&file_path) {
+        Ok(file) => file,
+        Err(err) => {
+            println!("Failed to open file '{}': {}", file_path, err,);
+            return;
+        }
+    };
+    let file = match unsafe { memmap2::Mmap::map(&file) } {
+        Ok(mmap) => mmap,
+        Err(err) => {
+            println!("Failed to map file '{}': {}", file_path, err,);
+            return;
+        }
+    };
+
+    match object::File::parse(&*file) {
+        Ok(object) => print_segments(&object),
+        Err(err) => {
+            println!("Failed to parse file segments '{}': {}", file_path, err,);
+            return;
+        }
+    };
+}
+
+fn print_segments(object: &object::File) {
+    for segment in object.segments() {
+        println!("-------------------------");
+        println!(
+            "name = {}",
+            segment.name().unwrap_or_default().unwrap_or_default()
+        );
+        println!("address = 0x{:016x}", segment.address());
+        println!("size = {}", segment.size());
+        println!("file range = {:?}", segment.file_range());
+        let rwe_flag = segment.rwe_flags();
+        println!("rwe flags = {}", rwe_flag);
+        if rwe_flag & SEGMENT_RWE_FLAG_READ > 0 {
+            println!("    - Read");
+        }
+        if rwe_flag & SEGMENT_RWE_FLAG_WRITE > 0 {
+            println!("    - Write");
+        }
+        if rwe_flag & SEGMENT_RWE_FLAG_EXECUTE > 0 {
+            println!("    - Execute");
+        }
```

Comment:
Instead of adding this example, can you give a `Debug` implementation for `SegmentFlags`, and then change the `Debug` implementation for `Segment` to display these flags. This will cause the `objdump` example to print out these flags.

Reply:
Also, updating the existing `objdump` tests might be a hassle, so it's fine if you prefer to not do these `Debug` changes, and I'll do them later.

Reply:
Yes, I found that:  if we update Debug implementation for Segment, there will be a lot of testing code and files need to be changed.
Since not all field of Segment is printed in Debug implementation.

```

impl<'data, 'file, R: ReadRef<'data>> fmt::Debug for Segment<'data, 'file, R> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        // It's painful to do much better than this
        let mut s = f.debug_struct("Segment");
        match self.name() {
            Ok(Some(ref name)) => {
                s.field("name", name);
            }
            Ok(None) => {}
            Err(_) => {
                s.field("name", &"<invalid>");
            }
        }
        s.field("address", &self.address())
            .field("size", &self.size())
            .finish()
    }
}
```

I think it is passable that we do not print flags in Debug implementation.